app/controllers/registrations_controller.rb

```diff
@@ -558,9 +558,24 @@ def stripe_webhook
   end
 
   def payment_completion
-    registration = Registration.includes(:competition).find(params[:id])
+    uses_v2 = params[:attendee_id].present?
+
+
+    registration = if uses_v2
+                     competition_id, user_id_s = params[:attendee_id].split("-")
+                     user_id = user_id_s.to_i
+                     MicroserviceRegistration.find_by(competition_id: competition_id, user_id: user_id)
+                   else
+                     Registration.includes(:competition).find(params[:id])
+                   end
     @competition = registration.competition
 
+    if registration.user.id == current_user.id
```

Comment:
I suppose that this should be `!==` instead?

Reply:
yup